src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -258,8 +258,13 @@ public void delete(final String eventTypeName) throws EventTypeDeletionException
             authorizationValidator.authorizeEventTypeAdmin(eventType);
 
             if (featureToggleService.isFeatureEnabled(EVENT_TYPE_DELETION_ONLY_ADMINS)) {
-                if (eventType.getAuthorization() == null || hasNonDeletableSubscriptions(eventType.getName())) {
```

Comment:
I was looking at this section and break my brain.

I think it makes sense to replace whole section
```
            if (featureToggleService.isFeatureEnabled(EVENT_TYPE_DELETION_ONLY_ADMINS)) {
                if (eventType.getAuthorization() == null || hasNonDeletableSubscriptions(eventType.getName())) {
                    throw new AccessDeniedException(eventType.asResource());
                }
            }

            if (featureToggleService.isFeatureEnabled(DELETE_EVENT_TYPE_WITH_SUBSCRIPTIONS)
                    || featureToggleService.isFeatureEnabled(EVENT_TYPE_DELETION_ONLY_ADMINS)) {
                topicsToDelete = deleteEventTypeWithSubscriptions(eventTypeName);
            } else {
                topicsToDelete = deleteEventTypeIfNoSubscriptions(eventTypeName);
            }
```

to something like
```
    if (eventType.getAuthorization() == null) {
        throw new AccessDeniedException(AuthorizationService.Operation.ADMIN, eventType.asResource(), "Can't delete event type that have no authorization specified");
    }

    if (featureToggleService.isFeatureEnabled(EVENT_TYPE_DELETION_ONLY_ADMINS)) {
        if (hasNonDeletableSubscriptions(eventType.getName())) {
            throw new AccessDeniedException(AuthorizationService.Operation.ADMIN, eventType.asResource(), "This event-type have subscriptions that cannot be deleted automatically.");
        }
        topicsToDelete = deleteEventTypeWithSubscriptions(eventTypeName);
    } else if (featureToggleService.isFeatureEnabled(DELETE_EVENT_TYPE_WITH_SUBSCRIPTIONS)) {
        topicsToDelete = deleteEventTypeWithSubscriptions(eventTypeName);
    } else {
        topicsToDelete = deleteEventTypeIfNoSubscriptions(eventTypeName);
    }
```

Reply:
I agree, I'll do it.